src/ray/core_worker/task_event_buffer.cc

```diff
@@ -211,6 +289,7 @@ bool TaskEventBufferImpl::Enabled() const { return enabled_; }
 
 void TaskEventBufferImpl::GetTaskStatusEventsToSend(
     std::vector<std::unique_ptr<TaskEvent>> *status_events_to_send,
+    std::vector<std::unique_ptr<TaskEvent>> *dropped_status_events_to_write,
```

Comment:
Instead of maintaining dropped tasks, can we just maintain a new vector just for export events? And we can just make it 10X original vector. I think maintaining another dropped tasks that are not dropped is not intuitive + it can easily screw up ordering depending on task drop policy.

Reply:
Sounds good. I needed to make the buffers for these events contain a `shared_ptr` rather than `unique_ptr` though so the same object can be moved to multiple buffers.

Reply:
before reviewing, can you just run microbenchmark to see if there's any noticeable regression? lmk if you don't know how to run them!

Reply:
yes, ran in https://buildkite.com/ray-project/release/builds/21683